shims/spark33/src/main/scala/org/apache/gluten/sql/shims/spark33/Spark33Shims.scala

```diff
@@ -231,6 +231,9 @@ class Spark33Shims extends SparkShims {
         case _ =>
       }
     }
+    metadataColumn.put(InputFileName().prettyName, file.filePath)
```

Comment:
If I understand it correctly, the semantics of generateMetadataColumns is to return the metadata columns according to the input `metadataColumnNames`, but we are adding these InputFile related metadata column without checking whether `metadataColumnNames` contains them?

Reply:
currently `metadataColumnsNames`  could only contains `vinalia spark` supported `constant` file metadta and `generated` filemetadata data.  `InputFileName` etc is not one of them.

Currently is fine to always put those 3 addtitional metadata columns as a default common metadata columns here which only used for the specific case of `project with input_file_name` similiar secnario.